src/components/tl/ucp/allgather/allgather.c

```diff
@@ -58,3 +58,37 @@ char *ucc_tl_ucp_allgather_score_str_get(ucc_tl_ucp_team_t *team)
                       UCC_TL_UCP_ALLGATHER_DEFAULT_ALG_SELECT_STR, algo_num);
     return str;
 }
+
+ucc_status_t loopback_self_copy(void *rbuf, void *sbuf, size_t data_size,
+                                ucc_memory_type_t rmem, ucc_memory_type_t smem,
+                                ucc_rank_t rank, ucc_tl_ucp_team_t *team,
+                                ucc_tl_ucp_task_t *task)
+{
+    ucc_status_t status;
+    status = ucc_tl_ucp_send_nb(sbuf, data_size, smem, rank, team, task);
```

Comment:
Instead of `status = ... ; if (UCC_OK != status) ...` you can use UCC_CHECK_GOTO(..., err)